lib/services.js

```diff
@@ -363,12 +363,24 @@ const services = {
                     }
 
                     if (!Array.isArray(objectMD.location)) {
-                        data.delete(objectMD.location, deleteLog);
-                        return cb(null, res);
+                        return data.delete(objectMD.location, deleteLog, err => {
+                            if (err) {
+                                log.warn('potential orphan in storage', {
+                                    object: objectMD.location,
+                                    error: err,
+                                });
+                                return cb(err);
+                            }
+                            return cb(null, res);
+                        });
```

Comment:
Note for reviewers: this code was not looking for any error when deleting data from storage. In case of error, that would create both orphans and invisible error codes for the user. Let me know if that should not be changed due to some non-documented reason(s).